dmoj/checkers/sorted.py

```diff
@@ -3,18 +3,29 @@
 import six
 from six.moves import map, zip, filter
 
+from dmoj.error import InternalError
 from dmoj.utils.unicode import utf8bytes
 
 
-def check(process_output, judge_output, **kwargs):
-    process_lines = list(filter(None, resplit(b'[\r\n]', utf8bytes(process_output))))
-    judge_lines = list(filter(None, resplit(b'[\r\n]', utf8bytes(judge_output))))
+def check(process_output, judge_output, split_on='lines', **kwargs):
+    split_pattern = {
+        'lines': b'[\r\n]',
+        'whitespace': b'[\r\n\s]',
+    }.get(split_on)
+
+    if not split_pattern:
+        raise InternalError('invalid `split_on` mode')
+
+    process_lines = list(filter(None, resplit(split_pattern, utf8bytes(process_output))))
```

Comment:
Consider just move the dict into global scope and do `re.compile` there instead of relying on caching behaviour.